internal/worker/containerprovisioner/containerprovisioner.go

```diff
@@ -265,17 +265,18 @@ func (p *containerProvisioner) getStartTask(ctx context.Context, harvestMode con
 	}
 
 	task, err := provisionertask.NewProvisionerTask(provisionertask.TaskConfig{
-		ControllerUUID:          controllerCfg.ControllerUUID(),
-		HostTag:                 hostTag,
-		Logger:                  p.logger,
-		HarvestMode:             harvestMode,
-		ControllerAPI:           p.controllerAPI,
-		MachinesAPI:             p.machinesAPI,
-		DistributionGroupFinder: p.distributionGroupFinder,
-		ToolsFinder:             p.toolsFinder,
-		MachineWatcher:          machineWatcher,
-		Broker:                  p.broker,
-		ImageStream:             modelCfg.ImageStream(),
+		ControllerUUID:               controllerCfg.ControllerUUID(),
+		HostTag:                      hostTag,
+		Logger:                       p.logger,
+		HarvestMode:                  harvestMode,
+		ControllerAPI:                p.controllerAPI,
+		MachinesAPI:                  p.machinesAPI,
+		GetMachineInstanceInfoSetter: machineInstanceInfoSetter,
```

Comment:
Eventually, we would want this to be passed in via the manifold, so that for controller agents, this would be identical to the compute provisioner. This is something we can do at a later date.

Reply:
I had originally done that, but then I changed my mind because IMO this is not public logic to be injected *into* the (provisioner) workers, it's instead pure internal implementation detail and some business logic (i.e. should I go through the API or hit the DB) and (again IMO) I think it's better to keep it inside.
If you don't agree I can do it, it's not an issue.